chore: Use `--no-cache-dir` flag to `pip` in Dockerfiles, to save space

rajpratik71 opened the pull request:
Using "--no-cache-dir" flag in pip install ,make sure dowloaded packages
by pip don't cached on system . This is a best practise which make sure
to fetch ftom repo instead of using local cached one . Further , in case
of Docker Containers , by restricing caching , we can reduce image size.
In term of stats , it depends upon the number of python packages
multiplied by their respective size . e.g for heavy packages with a lot
of dependencies it reduce a lot by don't caching pip packages.

Further , more detail information can be found at

https://medium.com/sciforce/strategies-of-docker-images-optimization-2ca9cc5719b6

Signed-off-by: Pratik Raj <rajpratik71@gmail.com>

parthea commented:
Hi @Rajpratik71 , Please could you resolve conflicts ?

parthea commented:
/gcbrun

parthea commented:
@Rajpratik71 The `conventionalcommits.org` check is failing. Please can you take a look? Changing the commit message to `chore: Use --no-cache-dir flag to pip in Dockerfiles, to save space` may resolve the issue.

rajpratik71 committed:
chore: Use `--no-cache-dir` flag to `pip` in Dockerfiles, to save space

Using "--no-cache-dir" flag in pip install ,make sure dowloaded packages
by pip don't cached on system . This is a best practise which make sure
to fetch ftom repo instead of using local cached one . Further , in case
of Docker Containers , by restricing caching , we can reduce image size.
In term of stats , it depends upon the number of python packages
multiplied by their respective size . e.g for heavy packages with a lot
of dependencies it reduce a lot by don't caching pip packages.

Further , more detail information can be found at

https://medium.com/sciforce/strategies-of-docker-images-optimization-2ca9cc5719b6

Signed-off-by: Pratik Raj <rajpratik71@gmail.com>

Rajpratik71 commented:
> @Rajpratik71 The `conventionalcommits.org` check is failing. Please can you take a look? Changing the commit message to `chore: Use --no-cache-dir flag to pip in Dockerfiles, to save space` may resolve the issue.

@parthea  done as required

parthea commented:
/gcbrun

parthea commented:
@yoshi-java @yoshi-go PTAL

parthea commented:
@googleapis/yoshi-java @googleapis/yoshi-go PTAL

parthea commented:
I'm going to close this PR as there is potential risk that downstream tests will fail and the risk is not worth the benefit of smaller images.
